yb-voyager/src/tgtdb/conn_pool.go

```diff
@@ -280,6 +281,32 @@ func (pool *ConnectionPool) shuffledConnUriList() []string {
 	return connUriList
 }
 
+func (pool *ConnectionPool) RemoveConnectionsForHosts(servers []string) error {
+	log.Infof("Checking for connections on host: %s", servers)
+	var conn *pgx.Conn
+	var gotIt bool
+	size := pool.size
+	for i := 0; i < size; i++ {
+		conn, gotIt = <-pool.conns
+		if !gotIt {
+			break
+		}
+
+		if conn == nil {
```

Comment:
we should continue here, not break. 
Also return the connection back to the pool